spec/models/member_spec.rb

```diff
@@ -30,66 +30,66 @@
 
 RSpec.describe Member do
   let(:user) { create(:user) }
-  let(:role) { create(:role) }
+  let(:project_role) { create(:role) }
+  let(:global_role) { create(:global_role) }
   let(:project) { create(:project) }
-  let(:member) { create(:member, user:, roles: [role]) }
-  let(:new_member) { build(:member, user:, roles: [role], project:) }
+  let(:work_package) { create(:work_package, project:) }
 
-  describe 'Associations' do
-    it { expect(member).to belong_to(:principal) }
-    it { expect(member).to have_many(:member_roles) }
-    it { expect(member).to have_many(:roles) }
+  subject(:member) { build(:global_member, user:, roles: [global_role]) }
+
+  describe 'relations' do
+    it { expect(member).to have_many(:member_roles).dependent(:destroy) }
+    it { expect(member).to belong_to(:project).optional }
+    it { expect(member).to have_many(:roles).through(:member_roles) }
+    it { expect(member).to belong_to(:principal).required }
+    it { expect(member).to belong_to(:entity).optional }
 
     it do
-      expect(member).to have_many(:oauth_client_tokens)
-        .with_foreign_key(:user_id)
-        .with_primary_key(:user_id)
-        .dependent(nil)
+      expect(member).to have_many(:oauth_client_tokens).with_foreign_key(:user_id).with_primary_key(:user_id).dependent(nil)
     end
   end
 
-  describe '#project' do
-    context 'with a project' do
-      it 'is valid' do
-        expect(new_member)
-          .to be_valid
-      end
-    end
-
-    context 'without a project (global)' do
-      let(:project) { nil }
+  describe 'validations' do
+    it { expect(member).to validate_uniqueness_of(:user_id).scoped_to(%i[project_id entity_type entity_id]) }
+    it { expect(member).to validate_inclusion_of(:entity_type).in_array(["WorkPackage"]).allow_blank }
```

Comment:
I continue to be surprised by the matchers provided by `type: :model`

Reply:
It is https://github.com/thoughtbot/shoulda-matchers